flow-plugins/vaadin-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinPlugin.kt

```diff
@@ -0,0 +1,66 @@
+/**
+ *    Copyright 2000-2020 Vaadin Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.vaadin.gradle
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.plugins.JavaPlugin
+import org.gradle.api.tasks.SourceSetContainer
+import org.gradle.api.tasks.bundling.Jar
+
+/**
+ * The main class of the Vaadin Gradle Plugin.
+ * @author mavi@vaadin.com
+ */
+public class VaadinPlugin : Plugin<Project> {
+    override fun apply(project: Project) {
+        // we need Java Plugin conventions so that we can ensure the order of tasks
+        project.pluginManager.apply(JavaPlugin::class.java)
+        var extensionName = "vaadin"
+        if (project.extensions.findByName(extensionName) != null) {
+            // fixes https://github.com/vaadin/vaadin-gradle-plugin/issues/26
+            extensionName = "vaadin14"
```

Comment:
```suggestion
            extensionName = "vaadinPlatform"
```
Should not be named 14 as it is not supported by this version.